openedx_tagging/core/tagging/rest_api/v1/views.py

```diff
@@ -423,9 +430,82 @@ class TaxonomyTagsView(ListAPIView):
         * 400 - Invalid query parameter
         * 403 - Permission denied
         * 404 - Taxonomy not found
+
+    **Create Query Parameters**
+        * pk (required) - The pk of the taxonomy to create a Tag for
+
+    **Create Request Body**
+        * tag (required): The value of the Tag that should be added to
+          the Taxonomy
+        * parent_tag_value (optional): The value of the parent tag that the new
+          Tag should fall under
+        * extenal_id (optional): The external id for the new Tag
+
+    **Create Example Requests**
+        POST api/tagging/v1/taxonomy/:pk/tags                                       - Create a Tag in taxonomy
+        {
+            "value": "New Tag",
+            "parent_tag_value": "Parent Tag"
+            "external_id": "abc123",
+        }
+
+    **Create Query Returns**
+        * 201 - Success
+        * 400 - Invalid parameters provided
+        * 403 - Permission denied
+        * 404 - Taxonomy not found
+
+    **Update Query Parameters**
+        * pk (required) - The pk of the taxonomy to update a Tag in
+
+    **Update Request Body**
+        * tag (required): The value (identifier) of the Tag to be updated
+        * updated_tag_value (required): The updated value of the Tag
+
+    **Update Example Requests**
+        PUT api/tagging/v1/taxonomy/:pk/tags                                        - Update a Tag in Taxonomy
+        {
+            "tag": "Tag 1",
+            "updated_tag_value": "Updated Tag Value"
+        }
+        PATCH api/tagging/v1/taxonomy/:pk/tags                                      - Update a Tag in Taxonomy
+        {
+            "tag": "Tag 1",
+            "updated_tag_value": "Updated Tag Value"
+        }
```

Comment:
Nit: I think it would be a bit more RESTful to put the old value in the URL and just use `value` as the name of the new value field. Though since `value` is a mutable ID it's not a huge deal. So no need to change for now.

What I _would_ definitely recommend we do now is remove the `PUT` example from the docstring. Because if we add the ability to change other fields like `external_id` via this API endpoint in the future, anyone who is using `PUT` to change the name will now be accidentally overwriting/deleting the `external_id`, since PUT means "completely replace the tag with this new spec". But `PATCH` explicitly means "update the fields I specify to the new values and leave others unchanged".

Reply:
I see, for sure, I removed the `PUT` example as suggested, and kept the endpoint as is for now.